### ISSUE TITLE
Is this project not being maintained/abandoned?

### ISSUE THREAD

**Issue by obj-obj** [first-rust-competition/first-rust-competition#100]
No commits since Dec 2020, and no new versions on crates.io since Jan 2019

**Comment by auscompgeek** [comment]
Hi! I'm not too sure about others, but work and other projects have been keeping me busy lately. I definitely don't have any plans to abandon this project.

Unfortunately this repo has bitrotted for long enough that there's probably a fair amount of work to get this updated for newer WPILib versions, but I think there is a solid foundation here to continue working off.

There are a few PRs lying around which I've been meaning to get around to merging…

**Comment by obj-obj** [comment]
Oh cool, thanks!
I might try contributing if I actually become good at coding in the near future

**Comment by obj-obj** [comment]
This is actually a pretty good project for learning FFI/bindgen and rust at the same time

**Comment by obj-obj** [comment]
I have started a separate project at https://gitlab.com/string_rs/wpilib_rs

**Comment by obj-obj** [comment]
Welp that didn't end well, I will do work this summer to perhaps have it working by next build season

**Comment by bobbbay** [comment]
@auscompgeek I'm interested in maintaining, if it helps. I want to run Rust on the field by next season. Let me know if/how I can help.

**Comment by ElectroMagneticInterference** [comment]
I am also willing to helps with it, I am not the best at rust but It would be awesome to get working.

**Comment by obj-obj** [comment]
If I can help as well ping me

**Comment by Supermath101** [comment]
Does anyone want to chat in the unofficial FRC Discord server about the future of this project?

**Comment by bobbbay** [comment]
I'd prefer to chat here if that's okay with everyone (I don't use Discord :)). What is there to chat about?

**Comment by auscompgeek** [comment]
If you folks want a more synchronous discussion, we do have a Gitter room: https://gitter.im/first-rust-competition/community

**Comment by AngleSideAngle** [comment]
I would also like to contribute to this project, likely with a few other members of my team's programming department. Currently we are waiting on #102 before attempting to add features.

**Comment by bobbbay** [comment]
@AngleSideAngle until #102 is merged, I want to continue my work (from scratch) at https://github.com/arfur-rs. I could use some helping hands for sure!

**Comment by AngleSideAngle** [comment]
@bobbbay Out of curiosity, why do you want to start a new project from scratch?

**Comment by bobbbay** [comment]
Great question. I had three main reasons:

 * I have zero connection to the admin team here. I have tried e-mailing them for crate name access, @-mentioning them here, and sending messages on the official Gitter, all to no avail. Even if I were to continue with this codebase, I would have to completely rename everything in order to publish to crates.io.
 * The codebase here is mostly a copy of [a 2017-ish codebase](https://github.com/robotrs/rust-wpilib). *A lot* has changed in the past five years, and many design choices simply don't make sense anymore. It would honestly be simpler to start from scratch than to migrate the existing design choices to the new one.
 * The code written here is *very* difficult to reproduce. It uses CLI calls instead of libraries, meaning Windows support will be very difficult. It also depends largely on your current environment - one out-of-version command line application - or in fact, one command line application with breaking changes in the future - will totally screw up this app, and won't be the easiest to diagnose.

In short, although this codebase has been instrumental in the journey towards Rust stabilization, it is an out-of-date monolith that would be saner to re-write.

**Comment by auscompgeek** [comment]
> Even if I were to continue with this codebase, I would have to completely rename everything in order to publish to crates.io.

This will become a problem on my end too... and the `wpilib` and `wpilib-sys` crate names would be good to keep using.

@Lytigas would you be able to add @first-rust-competition/core as an owner for the crates in this repo?

- [ ] cargo-frc
- [ ] wpilib
- [ ] wpilib-sys

**Comment by Lytigas** [comment]
Sorry for being radio silent on this continuation business; I've mostly left FRC behind while I'm in college. I'm glad people want to continue working on this. I'll add core ownership on crates like @auscompgeek suggested. Let me know if there's anything else I can or need to do.

**Comment by toastinthetub** [comment]
Is this dead?
